### PR TITLE
 feat: Handle empty bso body on /meta/global

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

This adds some additional debug around handling BSOs to try and identify
the source of the server returning 400 on PUT `/1.5/xxx/storage/meta/global`.
There may be a related client bug where it's sending no BSO on a
meta/global PUT, thus the very specific fix.

* also new clippy, new crap to clean up.

## Testing

Since we don't have a lot of tests that check full path, the test I've been using is to use `/tools/hawk/make_hawk_token.py` to generate a valid `Authorization` header

then call 
```
curl -v -X PUT "http://localhost:8000/1.5/1/storage/meta/global" \
    -H "content-type: application/json" \
    -H "Authorization: $AUTH"
```

## Issue(s)

Issue #342

### CODE REVIEW

**Comment by tublitzed** [comment]
When I run `python3 -m venv venv && venv/bin/pip install -r requirements.txt` in `tools/hawk` and then `python3 make_hawk_token.py`, I see this: ```  File "make_hawk_token.py", line 21, in <module>
    import hawkauthlib
ModuleNotFoundError: No module named 'hawkauthlib'``` - I'm guessing I'm missing some setup instruction for running that script?

**Comment by jrconlin** [comment]
@tublitzed Hrm. Interesting. 

Are you running python from inside of the virtualenv? (I'll add some additional instructions in the README to make sure that folks do that.)

**Comment by tublitzed** [comment]
@jrconlin - that was it. `source venv/bin/activate` fixed it

**Comment by tublitzed** [comment]
I just tried using both CURL and Postman, getting a 401:

Screen Shot 2019-12-04 at 1 52 43 PM

File: tools/hawk/README.md [hunk 1]
```diff
@@ -9,4 +9,16 @@ this will create a python virtual environment in the `/venv` directory.
 
 *Note* You may need to install `python3-venv` for the above to work.
 
-Use `-h` for help.
\ No newline at end of file
+Once the virtual env is installed, run `. venv/bin/activate`. This
```
Comment (tublitzed) [inline tools/hawk/README.md]: thanks for adding this!

**Comment by jrconlin** [comment]
I've updated the test instructions to duplicate the local testing I did. I will follow up with another Tools PR to include scripts.

**Pull request merged**